src/datadog/telemetry/telemetry_impl.cpp

```diff
@@ -244,27 +354,339 @@ void Telemetry::send_telemetry(StringView request_type, std::string payload) {
+      {"tracer_time", tracer_time},
+      {"runtime_id", tracer_signature_.runtime_id.string()},
+      {"debug", config_.debug},
+      {"application",
+       nlohmann::json::object({
+           {"service_name", tracer_signature_.default_service},
+           {"env", tracer_signature_.default_environment},
+           {"tracer_version", tracer_signature_.library_version},
+           {"language_name", tracer_signature_.library_language},
+           {"language_version", tracer_signature_.library_language_version},
+       })},
+      {"host",
+       {
+           {"hostname", host_info_.hostname},
+           {"os", host_info_.os},
+           {"os_version", host_info_.os_version},
+           {"architecture", host_info_.cpu_architecture},
+           {"kernel_name", host_info_.kernel_name},
+           {"kernel_version", host_info_.kernel_version},
+           {"kernel_release", host_info_.kernel_release},
+       }},
+  });
+}
+
+nlohmann::json Telemetry::generate_configuration_field(
+    const ConfigMetadata& config_metadata) {
+  // NOTE(@dmehala): `seq_id` should start at 1 so that the go backend can
+  // detect between non set fields.
+  config_seq_ids_[config_metadata.name] += 1;
+  auto seq_id = config_seq_ids_[config_metadata.name];
+
+  auto j = nlohmann::json{{"name", to_string(config_metadata.name)},
+                          {"value", config_metadata.value},
+                          {"seq_id", seq_id}};
+
+  switch (config_metadata.origin) {
+    case ConfigMetadata::Origin::ENVIRONMENT_VARIABLE:
+      j["origin"] = "env_var";
+      break;
+    case ConfigMetadata::Origin::CODE:
+      j["origin"] = "code";
+      break;
+    case ConfigMetadata::Origin::REMOTE_CONFIG:
+      j["origin"] = "remote_config";
+      break;
+    case ConfigMetadata::Origin::DEFAULT:
+      j["origin"] = "default";
+      break;
+  }
+
+  if (config_metadata.error) {
+    // clang-format off
+      j["error"] = {
+        {"code", config_metadata.error->code},
+        {"message", config_metadata.error->message}
+      };
+    // clang-format on
+  }
+
+  return j;
+}
+
 void Telemetry::capture_configuration_change(
     const std::vector<tracing::ConfigMetadata>& new_configuration) {
-  return tracer_telemetry_->capture_configuration_change(new_configuration);
+  configuration_snapshot_.insert(configuration_snapshot_.begin(),
+                                 new_configuration.begin(),
+                                 new_configuration.end());
+}
+
+void Telemetry::capture_metrics() {
+  std::time_t timepoint = std::chrono::duration_cast<std::chrono::seconds>(
+                              clock_().wall.time_since_epoch())
+                              .count();
+  for (auto& m : metrics_snapshots_) {
+    auto value = m.first.get().capture_and_reset_value();
+    if (value == 0) {
+      continue;
+    }
+    m.second.emplace_back(timepoint, value);
+  }
+}
+
+void Telemetry::log(std::string message, telemetry::LogLevel level,
```

Comment:
Out of curiosity, why did you remove the inline ?

Reply:
The main reason for this is that the definition is now in the `.cpp` file. If the question is why has it move in the `.cpp`, then, it's mainly to have all definitions in one place rather than scattered across the `.h` and `.cpp`. Do that answer your question?

Reply:
Yep, thank you